### PR TITLE
Merge develop 8.1 23.05.17

### PR DESCRIPTION
**Pull request by livecodeali**
Conflict in revmessageboxbehavior.livecodescript.

Kept both develop and develop-8.1 changes

### CODE REVIEW

**Comment by livecodeali** [comment]
@montegoulding There were three conflicts:
        Toolset/palettes/revpreferencesgui.rev
        Toolset/palettes/revresourcecenter.rev
        Toolset/palettes/script editor/revscripteditor.8.rev

I just ditched the changes from develop-8.1, so your script will have to be reapplied when this is merged :-)

File: tests/messagebox/execution.livecodescript [hunk 1]
```diff
@@ -163,22 +169,22 @@ on TestIntelligenceObjectMultipleCommand
...
-	ideExecuteScript "firstCommand; secondCommand", empty, tStack, false, tValidScript
+	ideExecuteScript "firstCommand; secondCommand", tStack, false, tValidScript
 	
 	TestAssert "intelligence object mutiple commands", the cValue of tStack is "value"
 end TestIntelligenceObjectMultipleCommand
 
 on TestDefaultStackCardScript
 	local tScript
-	put "on preOpenCard; set the cValue of this stack to" && quote & "value" & quote && "; end preOpenCard" into tScript
+	put "on preOpenCard; set the cValue of this stack of me to" && quote & "value" & quote && "; end preOpenCard" into tScript
```
Comment (montegoulding) [inline tests/messagebox/execution.livecodescript]: `this stack of me`?

File: Toolset/palettes/message box/behaviors/revmessageboxbehavior.livecodescript [hunk 2]
```diff
@@ -70,6 +70,18 @@ on openStack
    
    --if there is a stack currently being debugged then the debug mode is true, if no stack is currently being debugged then the debug mode is false
    revIDESetActiveStack
+
+   lock messages
+   set the revMessageBoxRedirect to the long id of me
```
Comment (montegoulding) [inline Toolset/palettes/message box/behaviors/revmessageboxbehavior.livecodescript]: this won't compile on `develop` as the property was removed.

**Comment by montegoulding** [comment]
OK here here's the stack if you want to apply it in this patch. Just click the button and it will search all stacks for legacy inks, change them appropriately based on what I found in the IDE and save any it changes.
[inks.livecode.zip](https://github.com/livecode/livecode-ide/files/1045973/inks.livecode.zip)

**Comment by livecodeali** [comment]
@montegoulding Does this look ok now?

**Comment by montegoulding** [comment]
@livecode-vulcan review ok 07fa960

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok 07fa960286ddf8cd36f6a9f3136a593df44ab5b7

**Pull request merged**